src/validator/is-port.ts

```diff
@@ -0,0 +1,46 @@
+import {STRule} from '../rule/rule';
+import {STRuleFn} from '../rule/fn';
+import {STRuleModifiers} from '../rule/modifiers';
+import {STRuleNode} from '../rule/node';
+import {STRuleNodeType} from '../rule/node-type';
+import {isInteger} from './is-integer';
+
+export type STOpIsPort<CallerType> = () => CallerType;
+
+export const isPort = (curr: number): boolean => {
+	if (typeof curr !== 'number') {
+		return false;
+	}
+
+	if (curr <= -1) {
```

Comment:
When using `<=`, think about what you're trying to communicate in the code. Here, our valid range is `0` to `65343`, right? Therefore, there's a meaningful reason to check values strictly falling outside that range. When using less than or equal to, think about what you want to communicate and why you'd want to say less than or equal to vs strictly less than than. For instance, for loops iterating in reverse communicate something important by saying `i >= 0`. As long as `i` is a positive integer, we're okay. Conversely, saying `i < 0` is communicating that you're referring to all negative values for i. Here, because your range is specific, I'd simply do a strict less than / greater than to check values inside or outside the range.

- [ ] Simplify to `curr < 0` to make it a little easier to understand.